### ISSUE TITLE
Null Reference Exception on RMB menu for pawn preparing for vehicle caravan

### ISSUE THREAD

**Issue by mwcrow** [SmashPhil/Vehicle-Framework#122]
**Describe the bug**
An NRE thrown when attempting to open a RMB menu on any item for pawn preparing for vehicle caravan (intent: access the 'add to caravan' context menu). 

```
Root level exception in OnGUI(): System.NullReferenceException: Object reference not set to an instance of an object
  at (wrapper dynamic-method) RimWorld.FloatMenuMakerMap.RimWorld.FloatMenuMakerMap.AddHumanlikeOrders_Patch6(UnityEngine.Vector3,Verse.Pawn,System.Collections.Generic.List`1<Verse.FloatMenuOption>)
  at (wrapper dynamic-method) RimWorld.FloatMenuMakerMap.RimWorld.FloatMenuMakerMap.ChoicesAtFor_Patch3(UnityEngine.Vector3,Verse.Pawn,bool)
  at RimWorld.FloatMenuMakerMap.TryMakeFloatMenu (Verse.Pawn pawn) [0x000dc] in <eeb29632ad124b59a4e69a77603ced84>:0 
  at (wrapper dynamic-method) RimWorld.Selector.RimWorld.Selector.HandleMapClicks_Patch2(RimWorld.Selector)
  at RimWorld.Selector.SelectorOnGUI () [0x00000] in <eeb29632ad124b59a4e69a77603ced84>:0 
  at RimWorld.MapInterface.HandleLowPriorityInput () [0x0000f] in <eeb29632ad124b59a4e69a77603ced84>:0 
  at (wrapper dynamic-method) RimWorld.UIRoot_Play.RimWorld.UIRoot_Play.UIRootOnGUI_Patch4(RimWorld.UIRoot_Play)
  at (wrapper dynamic-method) Verse.Root.Verse.Root.OnGUI_Patch2(Verse.Root)
UnityEngine.StackTraceUtility:ExtractStackTrace ()
(wrapper dynamic-method) Verse.Log:Verse.Log.Error_Patch6 (string)
(wrapper dynamic-method) Verse.Root:Verse.Root.OnGUI_Patch2 (Verse.Root)
```

Additional information from "VIsual Exceptions" mod: 

```
Exception in Verse.Root.OnGUI: System.NullReferenceException: Object reference not set to an instance of an object
  at RimWorld.FloatMenuMakerMap.AddHumanlikeOrders (UnityEngine.Vector3 clickPos, Verse.Pawn pawn, System.Collections.Generic.List`1[T] opts) [0x04a15] in <eeb29632ad124b59a4e69a77603ced84>:0 
     - transpiler SmashPhil.VehicleFramework: IEnumerable`1 Vehicles.CaravanFormation:AddHumanLikeOrdersLoadVehiclesTranspiler(IEnumerable`1 instructions, ILGenerator ilg)
     - prefix OskarPotocki.VFECore: Void VanillaWeaponsExpanded.FloatMenuMakerMap_AddHumanlikeOrders_Patch:Prefix(Pawn pawn)
     - postfix GeologicalLandformsMod.Main: Void GeologicalLandforms.Patches.Patch_RimWorld_FloatMenuMakerMap:AddHumanlikeOrders(Vector3& clickPos, Pawn& pawn, List`1& opts)
     - postfix OskarPotocki.VFECore: Void VanillaWeaponsExpanded.FloatMenuMakerMap_AddHumanlikeOrders_Patch:Postfix()
     - postfix OskarPotocki.VFECore: Void VFECore.Patch_FloatMenuMakerMap+AddHumanlikeOrders_Fix:Postfix(Vector3 clickPos, Pawn pawn, List`1& opts)
     - postfix legodude17.BetterBedAssign: Void BetterBedAssign.BetterBedAssignMod:AddOwnerAssign(Vector3 clickPos, Pawn pawn, List`1 opts)
     - postfix OskarPotocki.HeavyWeapons: Void HeavyWeapons.Patch_FloatMenuMakerMap+AddHumanlikeOrders_Fix:Postfix(Vector3 clickPos, Pawn pawn, List`1& opts)
  at RimWorld.FloatMenuMakerMap.ChoicesAtFor (UnityEngine.Vector3 clickPos, Verse.Pawn pawn, System.Boolean suppressAutoTakeableGoto) [0x000d1] in <eeb29632ad124b59a4e69a77603ced84>:0 
     - postfix OskarPotocki.VFECore: Void VFE.Mechanoids.HarmonyPatches.FloatMenuMakerMap_ChoicesAtFor_Patch:Postfix(List`1& __result, Vector3 clickPos, Pawn pawn, Boolean suppressAutoTakeableGoto)
     - postfix GiddyUp: Void GiddyUp.Harmony.FloatMenuMakerMap_ChoicesAtFor:Postfix(Vector3 clickPos, Pawn pawn, List`1 __result)
     - postfix rjw: List`1 rjw.RMB_Menu:ChoicesAtFor(List`1 __instance, Vector3 clickPos, Pawn pawn, Boolean suppressAutoTakeableGoto)
  at RimWorld.FloatMenuMakerMap.TryMakeFloatMenu (Verse.Pawn pawn) [0x000dc] in <eeb29632ad124b59a4e69a77603ced84>:0 
  at RimWorld.Selector.HandleMapClicks () [0x000f3] in <eeb29632ad124b59a4e69a77603ced84>:0 
     - prefix net.littlewhitemouse.LWM.DeepStorage: Boolean LWM.DeepStorage.Patch_HandleMapClicks:Prefix(Selector __instance, List`1 ___selected)
     - prefix SmashPhil.VehicleFramework: Boolean Vehicles.Extra:MultiSelectFloatMenu(List`1 ___selected)
  at RimWorld.Selector.SelectorOnGUI () [0x00000] in <eeb29632ad124b59a4e69a77603ced84>:0 
  at RimWorld.MapInterface.HandleLowPriorityInput () [0x0000f] in <eeb29632ad124b59a4e69a77603ced84>:0 
  at RimWorld.UIRoot_Play.UIRootOnGUI () [0x000f0] in <eeb29632ad124b59a4e69a77603ced84>:0 
     - prefix SmashPhil.SmashTools: Void SmashTools.MainMenuKeyBindHandler:HandleKeyInputs()
     - prefix Dubwise.PerformanceAnalyzer: Void Analyzer.H_KeyPresses:OnGUI()
     - postfix net.pardeike.rimworld.mod.camera+: Void CameraPlus.UIRoot_Play_UIRootOnGUI_Patch:Postfix()
     - postfix Krkr.RocketMan: Void RocketMan.KeyBinder:OnGUI()
  at Verse.Root.OnGUI () [0x00046] in <eeb29632ad124b59a4e69a77603ced84>:0 
     - transpiler net.pardeike.rimworld.lib.harmony: IEnumerable`1 VisualExceptions.ExceptionsAndActivatorHandler:Transpiler(IEnumerable`1 instructions, MethodBase original)
     - postfix UnlimitedHugs.HugsLib: Void HugsLib.Patches.Root_OnGUI_Patch:OnGUIHookUnfiltered()
     - postfix net.pardeike.rimworld.mod.camera+: Void CameraPlus.Root_OnGUI_Patch:Postfix()
```
 Admittedly there are few patches here, but I disabled non Vehicle Framework / Vanilla Expanded patches with same result. 
 
 My main issue is that pawns assigned to transport (haul) jobs, which should include 'pack vehicle' are not helping with loading. Only the driver and passenger are packing items to vehicle. Hopefully this has the same root cause. 

Log: 
https://gist.github.com/HugsLibRecordKeeper/5a4b2ea2e2bb7c9d2fa3e7593dfb8d55 
Some errors caused by removal of Achtung mod, which obfuscated the reported problem. 

**To Reproduce**
1. Using Vehicle Framework, VE:Vehicles and [FSF] Complex Jobs among other mods 
2. "Form Vehicle Caravan" from the world map option 
3. Caravan consists of single vehicle (Roadrunner from VE:Vehicles), driver, passenger and multiple cargo items. 
4. Caravan starts forming. Observe, that only driver and passenger are loading, despite pawns with priority on transport job ([FSF] Complex Jobs - it should be 'Haul' in vanilla. 'transport' includes "pack vehicle" job, according to tooltip) being available. 
5. Select driver or passenger, right click on any item - NFR is generated. 

**Additional context**
While I did spend decent amount of time to narrow down the issue, it is possible Vehicle Framework is not at the root cause here. However, due to my best understanding, it is a place to look at. Again the main issue is other pawns not helping with loading the caravan (which prevents usage of vehicles for trade caravans - we are talking few tons of goods in general), but I hope that the NRE here explains it.

**Comment by mwcrow** [comment]
Just a quick note that today's (17.XII) hotfix build didn't address this issue (at least for ongoing situation)

**Comment by SmashPhil** [comment]
There was no hotfix build.  Also, other colonists helping load the caravan is intended behavior.  Vanilla didn't used to have that QoL feature, so VF doesn't have that yet.  That's not a bug.  As for the exception, I'll look into it

**Comment by mwcrow** [comment]
Thank you for looking into this. I have just checked with Vanila (all DLC) + VE framework + Vehicles framework + VE:Vehicles and: 
1) Pawns assigned to haul job help loading cargo to animal caravan, even if they are not part of it
2) Pawns assigned to haul job help loading cargo to vehicle if 'load cargo' widget is used
3) Pawns assigned to haul job **do not** help loading cargo to vehicle caravan (which you stated is expected, but it does not appear like this is true for vanilla caravans) 

Moreover, I have noticed that the flow of forming caravan is different between two modes - in Vanilla caravan upon selecting 'form caravan' on the world map, user is initially required to select waypoints. For Vehicle Caravan the caravan formation window is  immediately opened, and than user is required to press 'change route'. Maybe this is intentional/required, so just sharing additional observation. 

Thank you for your time and your work in general.
